clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -394,15 +393,25 @@ public void failUnsentRequests(Node node, RuntimeException e) {
     private boolean trySend(long now) {
         // send any requests that can be sent now
         boolean requestsSent = false;
-        for (Map.Entry<Node, List<ClientRequest>> requestEntry: unsent.entrySet()) {
-            Node node = requestEntry.getKey();
-            Iterator<ClientRequest> iterator = requestEntry.getValue().iterator();
-            while (iterator.hasNext()) {
-                ClientRequest request = iterator.next();
-                if (client.ready(node, now)) {
-                    client.send(request, now);
-                    iterator.remove();
-                    requestsSent = true;
+        Set<Node> nodes = unsent.getNodes();
+        for (Node node: nodes) {
```

Comment:
Nit: there should be a space before `:`. There are a few more cases of this. We should change checkstyle to catch this issue.